components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -1118,6 +1120,100 @@ public OAuthRevocationResponseDTO revokeAuthzForAppsByResourceOwner(
+            }
+            if (activeDetailedTokens.size() > 0) {
+                String[] accessTokens = new String[activeDetailedTokens.size()];
+                Set<String> tenantDomains = new HashSet<String>();
+                tenantDomains.add(tenantDomain);
+                int countToken = 0;
+                for (AccessTokenDO detailToken : activeDetailedTokens) {
+                    String token = detailToken.getAccessToken();
+                    AuthenticatedUser authzUser = detailToken.getAuthzUser();
+
+                    accessTokens[countToken++] = token;
+                    // Add the tenant domains of authorized users since a tenant domain of a SAAS application could be
+                    // different from the user's tenant domain.
+                    tenantDomains.add(authzUser.getTenantDomain());
+
+                    OAuthCacheKey cacheKeyToken = new OAuthCacheKey(token);
+                    String scope = buildScopeString(detailToken.getScope());
+                    TokenBinding tokenBinding = detailToken.getTokenBinding();
+                    String tokenBindingReference = (tokenBinding != null &&
+                            StringUtils.isNotBlank(tokenBinding.getBindingReference())) ?
+                            tokenBinding.getBindingReference() : NONE;
+
+                    // Clear cache with AccessTokenDO.
+                    OAuthCache.getInstance().clearCacheEntry(cacheKeyToken);
+                    OAuthUtil.clearOAuthCache(consumerKey, authzUser, scope, tokenBindingReference);
+                    OAuthUtil.clearOAuthCache(consumerKey, authzUser, scope);
+                    OAuthUtil.clearOAuthCache(consumerKey, authzUser);
+                    OAuthUtil.clearOAuthCache(token);
+                }
+
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("Access tokens and token of users are removed from the cache for OAuth app in " +
+                            "tenant domain:" + tenantDomain + " with consumerKey: " + consumerKey);
+                }
+
+                // Revoking token from database.
+                try {
+                    OAuthTokenPersistenceFactory.getInstance().getAccessTokenDAO()
+                            .revokeAccessTokens(accessTokens, OAuth2Util.isHashEnabled());                } catch (IdentityOAuth2Exception e) {
+                    String errorMsg = "Error occurred while revoking access tokens for OAuth app in " +
+                            "tenant domain:" + tenantDomain + " with consumerKey: " + consumerKey;
+                    throw handleError(errorMsg, e);
+                }
+                // Revoking the oauth consent from database.
+                try {
+                    OAuthTokenPersistenceFactory.getInstance().getTokenManagementDAO().revokeOAuthConsentsByApplication(
+                            applicationName,
+                            tenantDomains.toArray(new String[0]));
```

Comment:
Why are we setting the size to 0?

Reply:
Then what's the purpose of collecting all the tenant domains?

Reply:
It's converting the hash set to a string array.